\documentclass[10pt]{article}
\usepackage[latin1]{inputenc}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{enumitem}
\usepackage{soul} % for strike-out

\usepackage{titlesec}
\titlespacing\section{0pt}{12pt plus 4pt minus 2pt}{0pt plus 2pt minus 2pt}

\usepackage[text={6.5in,8.5in},centering]{geometry}
\geometry{verbose,a4paper,tmargin=2.4cm,bmargin=2.4cm,lmargin=2.4cm,rmargin=2.4cm}

\author{Mark Novak \& Ben Dalziel}
\title{\emph{Course Proposal:}  Analytical Workflows\\\emph{\small{(temporarily} IB 599)}}
\date{}

\begin{document}

\maketitle

\section*{Course Motivation}
Graduate students are amassing ever larger datasets to ask ever more challenging questions with ever more sophisticated  statistical techniques. Many propose dissertation chapters entailing mathematical models and computational simulations. To keep pace requires students to straddle empirical, statistical, and theoretical approaches. These large-scale, multi-faceted, and ever more collaborative approaches lie at the heart of our department's \emph{integrative} approach to biology.

And yet, nowhere in our curriculum are students provided the basic tools and best practices with which to manage this complexity\footnote{See \emph{Related Courses at OSU} below.}. Moreover, these basic tools are standard practice in many of the career paths our students pursue.

Our course will provide students the tools they need to organize and manage their projects from inception to publication.  Working with their own data and/or models, students will learn and practice: (re)organizing projects into efficient, reproducible, and easily-modified ``\emph{analytical workflows}''; writing modular, transparent computer code for biology (in the programming language of their choice); tracking and communicating progress and hurdles (issues) to others using stand-alone and cloud-based version control and collaboration software.

\section*{Student-targeted Advertisement}
Have you proposed a modeling chapter for your dissertation but need support getting things up and running?  Are you sitting on a data set ready for analysis and visualization but don't know how or where to begin?  Maybe you're far along in some series of analyses and feel ``lost in the trees.''

This course will help you with these challenges by practicing the development and implementation of efficient, reproducible \emph{workflows} for your projects.  Every project should (and can) be modular and fully automated, hence reproducible, portable and easily modified.  Rerunning a model under a different set of parameters should (and can) be as simple as a few keystrokes. Regenerating all analyses, figures and tables after finding a typo in your code or dataset should (and can) be painless.

Efficient workflows start at project conception and end only if the project idea is itself a dead end.  Thus, in this course, we'll work to practice (1) refining and articulating project ideas and goals, (2) creating modular and automated analyses, and (3) using best-practices in coding and project management. We'll learn how to use Git, GitHub, \LaTeX\, and Markdown.  The instructors will mostly use \textsf{R} within RStudio, but users of other programming languages and text editors are welcome and encouraged.  You will need either (1) a large unwieldy dataset and an end goal (e.g., reproducing someone else's analysis or visualization) or (2) a dynamical model or simulation (or sufficiently well-developed ideas for one).  The use of other people's data or published models is also encouraged, as needed.

\section*{Course Catalog Description}
Theory and implementation of efficient, reproducible workflows -- including best-practices in scientific programming, project management, and collaboration --  for computational, analytical, and data-driven biological research.

\section*{Student Testimonials}
A preliminary IB599 version of this course was taught by Ben Dalziel (with support from Mark Novak) in the Spring of 2019.  Student eSET scores for this course were 6.0 (versus the departmental median of 4.9) and 6.0 (versus the departmental median of 5.1) out of 6.0 points possible for ``the course as a whole'' and ``the instructor's contribution'' respectively. The following are excerpts of the feedback that was received via email and the eSET evaluations.

\begin{quote}
	\emph{``I am of the opinion that this Analytical Workflow class is indispensable. Engaging in research comes with a massive deluge of papers, files, data, output, etc, and yet prior to this class I had never before been exposed to organizational ``best practices'' recommendations.''}
\end{quote}
\begin{quote}
	\emph{``After taking the class, [...] I feel empowered to solve issues with my model on my own.''}
\end{quote}
\begin{quote}
	\emph{``I'm so happy this class exists, and it was instrumental to much of the progress I've made this term.''}
\end{quote}
\begin{quote}
\emph{``Ben and Mark helped turn a daunting task that I'd been putting off (my modeling chapter) into a well-organized reality! Five stars.''}
\end{quote}
\begin{quote}
	\emph{``This class has been extremely useful for the conceptual organization and technical execution of my research. Thank you for organizing and leading this class!''}
\end{quote}
\begin{quote}
	\emph{``I really appreciate that you shared your experience and learning process with us (metacognitive reflection! best teaching practices!).''}
\end{quote}
\begin{quote}
	\emph{``I wasn't experienced in R/Git or savvy about best practices in coding or reproducible workflows. Ben and Mark helped me structure my independent work and provided essential information about programs and best practices in coding.''}
\end{quote}


\section*{Details}
\begin{tabular}{rl}
	\emph{Credits:}	&  4\\
	\emph{Frequency:} & Annually (starting \st{2021} 2020)\\
	\emph{Quarter:} & \st{Winter} Fall\\
	\emph{Course times:} &  Tuesday \& Thursday 10:00-11:50am\\
	\emph{Instructors:} & Mark Novak \& Ben Dalziel (of record alternating annually)\\
	\emph{Prerequisites}: & Graduate standing, or by instructor permission\\
	\emph{Location}: & TBD (likely ``remote'')
\end{tabular}

\section*{Learning Outcomes}
After successful completion of this course, students will be able to:
\begin{enumerate}
	\itemsep0em
	\item Translate a research plan into an explicit analytical workflow;
	\item Apply best practices in scientific programming to construct reproducible research;
	\item Manage and collaborate on complex research projects using a version control system;
	\item Apply analytical workflows to advance their dissertation research.
\end{enumerate}


\section*{Schedule}
\begin{enumerate}[label=\bfseries Week \arabic*:,leftmargin=*,labelindent=1em]
	\itemsep0em
	\item Course organization \& Philosophy of workflows
	\item Student workflow diagrams \& Project presentations
	\item Project setup \& Version control (Git)
	\item Team troubleshooting
	\item Collaborative projects, management \& debugging (GitHub)
	\item Coding best practices \& grammar
	\item Visualization theory \& workflows
	\item Writing notes, reports \& papers (\LaTeX, Markdown \& reference management)
	\item Team troubleshooting
	\item Project presentations
\end{enumerate}



\section*{Related courses at OSU}
OSU's existing \emph{statistics} courses focus exclusivively on data visualization.
Existing \emph{omics} courses introduce students to specific \emph{analysis pipelines} in particular programming environments, but do not address issues relating to project management as a whole.
Existing courses are also not relevant to the many IB students pursuing non-\emph{omic} research.

\noindent
\textbf{ST 537 - Data Visualization} \emph{E-campus only}\\
``Perceptual principles for displaying data; critique and improvement of data visualizations; use of color in visualization; principles of tidy data; strategies for data exploration; select special topics. \emph{Prerequisites: ST 512 or ST 517 or ST 552}''

\noindent
\textbf{BB 485/585 - Applied Bioinformatics}\\
 \emph{Prerequisites: BB 314 or BB 314H}''
``Fundamental concepts needed to understand the software and methods used in bioinformatics. Includes contemporary techniques such as databases, gene and genome annotations, functional annotations, sequence alignment, motif finding, secondary structure prediction, phylogenetic tree construction, high-throughput sequence data, ChIP-Seq peak identification, transcriptome profiling by RNA-Seq, microRNA discovery and target prediction.
\end{document}
